app/src/main/java/com/example/dkirilova/gym/activities/MainActivity.java

```diff
@@ -222,7 +192,63 @@ public void onClick(View view) {
         });
     }
 
-    public void setiGymController(GymFragment.IGymController iGymController) {
+    public void openMapFragment() {
+        configureToolbar(true, R.menu.map_menu);
+        openFragment(new GMapFragment(), null, null);
+    }
+
+    public void openFragment(Fragment fragment, Serializable serializable, String str) {
+
+        if (serializable != null && str != null) {
+            Bundle bundle = new Bundle();
+            bundle.putSerializable(str, serializable);
+            fragment.setArguments(bundle);
+        }
+        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
+        ft.replace(R.id.fragmentContainer, fragment).commit();
+    }
+
+    public void setIGymController(GymFragment.IGymController iGymController) {
         this.iGymController = iGymController;
     }
+
+    public void setIMapController(GMapFragment.IMapController iMapController) {
+        this.iMapController = iMapController;
+    }
+
+    public void setIGymDetailsController(GymDetailsFragment.IGymDetailsController iGymDetailsController) {
+        this.iGymDetailsController = iGymDetailsController;
+    }
+
+    public void setIExerciseController(IExerciseController iExerciseController) {
+        this.iExerciseController = iExerciseController;
+    }
+
+    public void editOrDelete(final Gym gym, final Exercise exercise) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("Edit or delete");
+        builder.setPositiveButton("EDIT", new DialogInterface.OnClickListener() {
```

Comment:
move hardcoded strings to the strings.xml resource file
e.g. "EDIT", "DELETE", "Edit or delete"